src/main/scala/tech/cryptonomic/conseil/routes/DataHelpers.scala

```diff
@@ -20,11 +20,26 @@ trait DataHelpers extends QueryStringListsServer with Validation with akkahttp.s
   import io.circe.syntax._
 
   /** Function validating request for the query endpoint */
-  override def validated[A](response: A => Route, invalidDocs: Documentation): Either[List[QueryValidationError], A] => Route = {
+  override def validated(response: QueryResponseWithOutput => Route, invalidDocs: Documentation): Either[List[QueryValidationError], QueryResponseWithOutput] => Route = {
```

Comment:
Why are we foregoing parametricity to use a concrete type?

Wouldn't pattern matching allow to be specific all the same?

Reply:
pattern matching was tricky in this case, but I've managed to find workaround - apparently I couldn't match `case Right(success@QueryResponseWithOutput(_, OutputType.json))` and return `response(success)` - compiler told me that the type is wrong. But when I've moved it as a last case and made it `case Right(success@QueryResponseWithOutput)` it compiles correctly.